### PR TITLE
Job context implementation

### PR DESCRIPTION
**Pull request by samplise**
### What changes were proposed in this pull request?

Wrap up job context into the object JobContext.

### Why are the changes needed?

Ensure the job context is accessed with control(Avoid implicit object update operations whenever possible
).

### Does this PR introduce any user-facing change?

No.

### How was this patch tested?

Unit test

### CODE REVIEW

File: dlrover/python/common/grpc.py [hunk 1]
```diff
@@ -506,3 +506,19 @@ class ElasticRunConfigRequest(Message):
 @dataclass
 class ElasticRunConfig(Message):
     configs: Dict[str, str] = field(default_factory=dict)
+
+
+@dataclass
+class SucceededRequest(Message):
+    pass
```
Comment (BalaBalaYi) [inline dlrover/python/common/grpc.py:513]: remove

File: dlrover/python/diagnosis/common/constants.py [hunk 2]
```diff
@@ -32,7 +36,11 @@ class DiagnosisDataType(object):
     XPU_TIMER_METRIC = "XPU_TIMER_METRIC"
 
 
-class DiagnosisAction(object):
+class DiagnosisActionType(object):
     NO_ACTION = "no_action"
     RESTART_WORKER = "restart_worker"
     RELAUNCH_WORKER = "relaunch_worker"
+    MASTER_RELAUNCH_WORKER = "master_relaunch_worker"
+    EVENT = "event"
```
Comment (BalaBalaYi) [inline dlrover/python/diagnosis/common/constants.py:44]: lint

File: dlrover/python/diagnosis/common/diagnosis_action.py [hunk 3]
```diff
@@ -0,0 +1,208 @@
...
+        return self._reason
+
+
+def is_same_action(action1: DiagnosisAction, action2: DiagnosisAction) -> bool:
+    return False
+
+
+class DiagnosisActionQueue:
+    def __init__(self):
+        self._actions: Dict[int, List[DiagnosisAction]] = {}
```
Comment (BalaBalaYi) [inline dlrover/python/diagnosis/common/diagnosis_action.py:160]: Should use limited 'deque' instead of 'list'.

**Review by samplise — COMMENTED** [review]
lgtm

**Review by BalaBalaYi — APPROVED** [review]

**Pull request merged**